chaste_codegen/model_with_conversions.py

```diff
@@ -0,0 +1,502 @@
+import logging
+
+import cellmlmanip
+from cellmlmanip.model import DataDirectionFlow
+from cellmlmanip.rdf import create_rdf_node
+from cellmlmanip.units import UnitStore
+from pint import DimensionalityError
+from sympy import (
+    Derivative,
+    Eq,
+    Expr,
+    Float,
+    Function,
+    Pow,
+    Wild,
+    log,
+    simplify,
+    sympify,
+)
+from sympy.codegen.cfunctions import log10
+from sympy.codegen.rewriting import ReplaceOptim, log1p_opt, optimize
+
+from chaste_codegen import LOGGER
+from chaste_codegen._rdf import OXMETA, PYCMLMETA, get_variables_transitively
+
+from ._math_functions import MATH_FUNC_SYMPY_MAPPING
+
+
+MEMBRANE_VOLTAGE_INDEX = 0  # default index for voltage in state vector
+CYTOSOLIC_CALCIUM_CONCENTRATION_INDEX = 1  # default index for cytosolic calcium concentration in state vector
+
+STIM_PARAM_TAGS = (('membrane_stimulus_current_amplitude', 'uA_per_cm2', True),
+                   ('membrane_stimulus_current_duration', 'millisecond', True),
+                   ('membrane_stimulus_current_period', 'millisecond', True),
+                   ('membrane_stimulus_current_offset', 'millisecond', False),
+                   ('membrane_stimulus_current_end', 'millisecond', False))
+
+
+# Optimisations to be applied to equations
+_V, _W = Wild('V'), Wild('W')
+# log(x)/log(10) --> log10(x)
+_LOG10_OPT = ReplaceOptim(_V * log(_W) / log(10), _V * log10(_W), cost_function=lambda expr: expr.count(
+    lambda e: (  # cost function prevents turning log(x) into log(10) * log10(x) as we want normal log in that case
+        e.is_Pow and e.exp.is_negative  # division
+        or (isinstance(e, (log, log10)) and not e.args[0].is_number))))
+# For P^n make sure n is passed as int if it is actually a whole number
+_POW_OPT = ReplaceOptim(lambda p: p.is_Pow and (isinstance(p.exp, Float) or isinstance(p.exp, float))
+                        and float(p.exp).is_integer(),
+                        lambda p: Pow(p.base, int(float(p.exp))))
+
+
+def load_model_with_conversions(model_file, use_modifiers=True, quiet=False):
```

Comment:
I would have thought `False` was a better default for `use_modifiers`.

Reply:
good point